rustls/src/lib.rs

```diff
@@ -512,14 +511,26 @@ mod test_macros;
 /// of rustls targetting architectures without atomic pointers to replace the implementation
 /// with another implementation such as `portable_atomic_util::Arc` in one central location.
 mod sync {
-    #[cfg(unstable_portable_atomic_arc)]
+    // Arc alias as exported by public API below
+    pub(crate) use crate::Arc;
+}
+
+mod arc_alias {
+    // NOTE that unsupported_std_arc_cfg option is NOT DOCUMENTED and NOT SUPPORTED - primary purpose is for extra testing
+    #[cfg(not(unsupported_std_arc_cfg))]
     #[allow(clippy::disallowed_types)]
     pub(crate) type Arc<T> = portable_atomic_util::Arc<T>;
-    #[cfg(not(unstable_portable_atomic_arc))]
+    #[cfg(unsupported_std_arc_cfg)]
```

Comment:
rename cfg option to something like `unstable_use_arc_from_stdlib` & document briefly in __Crate cfg options__ section (does not need to be supported)